File: Dev/unitTest_Engine_cpp_gtest/Graphics/PostEffect/PostEffectLightBloom.cpp
#include <Altseed.h>
#include <gtest/gtest.h>
#include <memory>
#include "../../EngineTest.h"


class Graphics_PostEffectLightBloom : public EngineTest
{
public:
	Graphics_PostEffectLightBloom(bool isOpenGLMode) :
		EngineTest(asd::ToAString("PostEffectLightBloom"), isOpenGLMode, 10)
	{}
protected:

	void OnStart() override
	{
		auto scene = std::make_shared<asd::Scene>();
		auto layer = std::make_shared<asd::Layer2D>();
		auto obj= std::make_shared<asd::TextureObject2D>();
		scene->SetHDRMode(true);
		scene->AddLayer(layer);
		layer->AddObject(obj);
		asd::Engine::ChangeScene(scene);

		auto g = asd::Engine::GetGraphics();
		auto texture = g->CreateTexture2D(asd::ToAString("Data/Texture/Sample1.png").c_str());
		obj->SetTexture(texture);
		obj->SetScale(asd::Vector2DF(1, 1));

		auto pe = std::make_shared<asd::PostEffectLightBloom>();
		pe->SetIntensity(5.0f);
		pe->SetThreshold(0.2f);
		pe->SetExposure(3.0f);
		layer->AddPostEffect(pe);
	}
};

ENGINE_TEST(Graphics, PostEffectLightBloom)

File: Dev/unitTest_Engine_cs/Graphics/PostEffect/PostEffectLightBloom.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics.PostEffect
{
	class PostEffectLightBloom : EngineTest
	{
		public PostEffectLightBloom()
			: base(10)
		{
		}

		protected override void OnStart()
		{
			var scene = new ace.Scene();
			var layer = new ace.Layer2D();
			var obj = new ace.TextureObject2D();
			scene.AddLayer(layer);
			layer.AddObject(obj);
			ace.Engine.ChangeScene(scene);

			var g = ace.Engine.Graphics;
			var texture = g.CreateTexture2D("Data/Texture/Sample1.png");
			obj.Texture = texture;
			obj.Scale = new ace.Vector2DF(1, 1);

			var pe = new ace.PostEffectLightBloom();
			pe.Intensity = 5.0f;
			pe.Threshold = 0.3f;
			pe.Power = 2.0f;
			layer.AddPostEffect(pe);
		}
	}
}
